### PR TITLE
[WIP] Documentation Improvements for SC16

### PR DESCRIPTION
**Pull request by citibeth**
For now, I'm putting all documentation changes in this PR.

### CODE REVIEW

File: lib/spack/docs/getting_started.rst [hunk 1]
```diff
@@ -35,11 +55,44 @@ For a richer experience, use Spack's `shell support
...
+
+Therefore, it is recommended that Spack users run with a *clean
+environment*, especially for ``PATH``.  Only software that comes with
+the system, or that you know you wish to use with Spack, should be
+included.  This procedure will avoid many strange build errors that no
+one knows how to fix.
+
+
+Although Spack will work as soon as you clone it, it won't necessarily
+be able to install any packages.  That is because Spack relies
```
Comment (davydden) [inline lib/spack/docs/getting_started.rst]: unfinished sentence.

File: lib/spack/docs/application_developer_suppport.rst [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+provides the benefits of having a simpler and traditional layout which
+may be used without any particular knowledge that its packages were
+built by Spack.
+
+Views can be used for a variety of purposes including:
+
+- A central installation in a traditional layout, eg ``/usr/local`` maintained over time by the sysadmin.
+- A self-contained installation area which may for the basis of a top-level atomic versioning scheme, eg ``/opt/pro`` vs ``/opt/dev``.
+- Providing an atomic and monolithic binary distribution, eg for delivery as a single tarball.
+- Producing ephemeral testing or developing environments.
```
Comment (davydden) [inline lib/spack/docs/application_developer_suppport.rst]: :+1: that's how i use views.

**Comment by davydden** [comment]
LGTM

**Comment by adamjstewart** [comment]
It looks like this PR is mostly just moving sections around to better locations. That's fine, but in #1576 I'm greatly modifying all of these sections. Can you wait until after #1576 is merged and then rearrange things with the new documentation? Otherwise I'm going to have to do everything all over again.

**Comment by citibeth** [comment]
This PR is a lot more than rearranging sections, that is just the start
(see WIP above).  I don't think it's reasonable to wait until #1576 is
merged because we need this documentation overhaul "yesterday."  Why don't
I take the doc changes you've made so far in #1576 and incorporate into
this PR?  #1576 then becomes a docs-free PR.

@tgamblin, your thoughts on this proposal?

**Comment by adamjstewart** [comment]
@citibeth #1576 has around 4,000 lines changed, and they're almost exclusively in the documentation... It's more of a documentation overhaul and then adding testing to prevent us from breaking it again.

**Comment by citibeth** [comment]
Adam,

How about this workflow proposal (or some variant thereof): you finish
rebasing today and then "pass it off" to me (I'm offline right now).  I
re-do the rearrange changes I did last night on your branch and check back
in while you're offline.  Once we complete those two steps, we should have
a version we can both work on.

Where do we host this shared branch?  I'm happy to make you a contributor
on my Git account; or you can make me a contributor on yours.

Does this sound reasonable to you?  We need to find some sort of
concurrency in this process, I don't think either of us can or should
"lock" the documentation for long.  I live in GMT-4 right now.

-- Elizabeth

**Comment by tgamblin** [comment]
@citibeth @adamjstewart: Looking at these, are there so many major conflicts?  @citibeth's stuff is mostly text. @adamjstewart's is mostly formatting.  And @citibeth's stuff is only in 5 files.  If we merge #1576, I don't mind rebasing this one on it...

**Comment by citibeth** [comment]
A bit more color here... most of the changes I need to do involve adding
new content.  If I can rearrange the versions of the sections as you have
them now, I don't think we'll be stepping on each others' toes too much.

**Comment by tgamblin** [comment]
@citibeth: I suspect if we merge @adamjstewart's first it won't take much to merge in on top of it.

**Comment by tgamblin** [comment]
@citibeth are you ok w/that?

**Comment by adamjstewart** [comment]
@tgamblin It may take me a while to rebase. I can start now and see how far I get by tomorrow? I don't have any problems with adding new documentation, but if you move documentation, then all of the changes I made to that section disappear.

**Comment by citibeth** [comment]
Am I right that although SC16 deadline is tomorrow, we have longer to
finish changes we want to the manual?  Because that lives on-line, not on
the USB keys we hand out... or are we planning on handing out a frozen
version of the manual?

In hindisght, my rearrangements consist mostly of moving certain sections
out of files they're currently in, and into new files I'm creating.  It's
not so many segments, just 2 or 3.  Once #1576 is merged, I will "rebase"
my work as follows:

1. I'll re-do the moves, into blank files, this time using the latest
version of the sections as I get them from #1576.  Make a commit.

2. I'll then merge from the branch I'm currently working on, which will
bring in all my new text.

Git is great.  I think this problem is resolved.  #1576 will merge first,
and then I'll deal.

-- Elizabeth

**Comment by adamjstewart** [comment]
@citibeth Alright, I'll finish up #1576 as soon as I can.

**Comment by tgamblin** [comment]
hmmmm... 63 commits?  I think you want to `rebase --onto develop 918cb16 HEAD`

**Comment by davydden** [comment]
@citibeth https://github.com/LLNL/spack/pull/1576 is already merged, just rebase on `develop`. You also have a lot of unneeded commits from Jul29. Make sure you fetch `<upatream>` and also that you rebase on `<upstream>/develop`.

**Comment by tgamblin** [comment]
`--onto` probably isn't necessary here so maybe ignore my previous comment.  Plain old `git rebase develop` should work.

**Pull request closed without merging**